log/hooks.go

```diff
@@ -0,0 +1,72 @@
+package log
+
+import (
+	"fmt"
+	bs "github.com/bugsnag/bugsnag-go"
+	"github.com/docker/machine/log/bugsnag"
+	"github.com/docker/machine/version"
+	"net"
+	"os"
+	"runtime"
+	"strings"
+)
+
+// Empty logger for bugsnag
+type void struct {
+}
+
+func (d *void) Printf(fmtString string, args ...interface{}) {
+}
+
+// SetBugsnag configures the bugsnag hook for logrus
+func SetBugsnag(apiKey string) error {
+	stage := fmt.Sprintf("%s (%s)", runtime.GOOS, runtime.GOARCH)
+
+	bs.Configure(bs.Configuration{
+		APIKey: apiKey,
+		// XXX we need to abuse bugsnag metrics to get the OS/ARCH information as a usable filter
+		// Can do that with either "stage" or "hostname"
+		ReleaseStage: stage,
+		//		Hostname: stage,
+		ProjectPackages: []string{"github.com/docker/machine/[^v]*"},
+		AppVersion:      version.Version + " (" + version.GitCommit + ")",
+		Logger:          new(void),
+		Synchronous:     true,
+	})
+
+	hook, err := bugsnag.NewBugsnagHook()
+	if err != nil {
+		return err
+	}
+
+	// Add runtime informations
+	hook.Add("app", "compiler", fmt.Sprintf("%s (%s)", runtime.Compiler, runtime.Version()))
+
+	hook.Add("device", "os", runtime.GOOS)
+	hook.Add("device", "arch", runtime.GOARCH)
+
+	// Add environ
+	for _, v := range os.Environ() {
+		idx := strings.Index(v, "=")
+		k := v[:idx]
+		v = v[idx+1:]
+		hook.Add("environ", k, v)
+	}
+
+	// Add network information
+	t, _ := net.Interfaces()
```

Comment:
Yes, and the program should continue working as normal in that case -- but what I was referring to is if the `net.Interfaces` call itself failed.  It could return `err` that we shouldn't swallow.  e.g. a variety of `syscall` errors in the function here https://golang.org/src/net/interface_bsd.go#L18